### ISSUE TITLE
Screen Reader: Global: The buttons with chevrons are announced without a rol

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77563]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Profile detail buttons with chevron (e.g., Display name) and note how they are announced

This is a global issue affecting all buttons with chevron across the ap, e.g.:
1. On Status screen
2. On Status - Clear After: the Date and Time
3. On Preferences
4. On Group Chat: the button of group name
5. On Profile - Contact Methods
6. On Profile - Address
7. On Profile - Timezone
8. On Security
9. On About
10. On Wallet - Add Bank Account
11. On Create Expense - Confirm Details
12. On Save the world
13. Filter - Date
14. On Filters
15. On Report Details

## Expected Result:
Each control should be announced with a button role.

## Actual Result:
The buttons with chevrons are announced without a role.

## Area issue was found in:
Profile

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Update the programmatic role of the buttons to ensure they are announced as such, using accessibilityRole="button".
https://appt.org/en/docs/react-native/samples/accessibility-role
https://reactnative.dev/docs/accessibility#accessibilityrole

## Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

<img width="1179" height="3575" alt="Image" src="https://github.com/user-attachments/assets/be6dc94b-b668-4173-9e2c-c0097a9fadb0" />

<img width="1179" height="2430" alt="Image" src="https://github.com/user-attachments/assets/ab7c4db8-e294-44ee-b533-8f8c5268052f" />

https://github.com/user-attachments/assets/35ac3343-0368-4115-a630-4475aaa6b0e7

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005909
**Issue reported by:** NA
**Slack conversation:** NA
